### PR TITLE
New shuttle walls, doors and windows.

### PR DESCRIPTION
**Pull request by Bm0n**
## What Does This PR Do
Ports a new plastinum wall sprite(aka shuttle walls) from TG.

Ports matching shuttle airlock from TG.

Ports TG shuttle windows, also matching.

Reverts shuttle windows smoothing with shuttle walls(includes plastitanium type), it was added in PR https://github.com/ParadiseSS13/Paradise/pull/16620 as an undocumented change and has only made our shuttles look ugly. Block type non-slanted windows have no place smoothing with walls.

## Why It's Good For The Game
Updated shuttle sprites to help make our very old crummy looking shuttles better.

Basically shuttle with RTX on.
## Images of changes
This thing
image

Replaces this soul fueled machine
image

## Testing

I shot a bunch of shit with an L6 to test the new mask overlays for the shuttle doors

## Changelog
:cl: Bmon, Onule
imageadd: Ported new shuttle walls, windows and airlocks  from TGstation.
/:cl:

### CODE REVIEW

**Comment by Coolrune206** [comment]
I do like how the outsides of the walls look, especially on the diagonals, but that inside strip seems a little too dark compared to the rest of the wall.

**Comment by CinnamonSnowball** [comment]
The sprites do need an upgrade, but I'm really not sure this is it. Personally, *current* sprites look like an upgrade to *these*.

The new wall sprites clash *with themselves* via that dark inside strip, and I'm not sure whether it's poor quality of the screenshot, but they look really messy and noisy especially at the diagonal parts. They also arguably look even less "sloped" than current sprites, straying further from our sloped regular walls.
The windows also don't really seem to connect to the walls anymore, which is somewhat jarring and kind of the opposite of the treatment regular windows all over the station recently got.

**Comment by procdrone** [comment]
i agree these sprites dont look good.

**Comment by Bm0n** [comment]
> I do like how the outsides of the walls look, especially on the diagonals, but that inside strip seems a little too dark compared to the rest of the wall.

It adds a lot more colour variation to the homogonous white blob that was the old sprite. Keep in mind these walls are made of titanium + iron

Moving on...

> The new wall sprites clash _with themselves_ via that dark inside strip and very harsh differences between dark and very light grays in general

There is absolutely no clash between white and soft blue grey which is used throughout the sprite and not just in the middle. I do not see what you're talking about.

>and I'm not sure whether it's poor quality of the screenshot,

If you're ever in doubt go check icon diff and the sprite file. It's a big shuttle and I needed to get it all in frame, of course the quality dips when you stretch things out.

>but they look really messy and noisy especially at the diagonal parts. They also arguably look even less "sloped" than current sprites, straying further from our sloped regular walls.

It's a 32x32 sprite, there's limitations when angling them, these same limitations are also seen in the old sprite. Also I really hope you understand that **_only shuttle walls can slope, regular walls do not slope_**, so I really don't know what you're trying to argue here.

>The windows also don't really seem to connect to the walls anymore, which is somewhat jarring and kind of the opposite of the treatment regular windows all over the station recently got.

The shuttle window sprite was intended not to smooth with walls like how our old windows were. The current new regular windows are so they look good smoothed. Just because one sprite does it doesn't mean all of them should or that it'd look good.


Finally...

> i agree these sprites dont look good.

This is not feedback, I can't reply to this if you don't actually tell me what you don't like about the sprite.


Even though you all should have done this yourself(and you still should) here are some full res pictures of the walls:

<details><summary>NEW</summary>

image

image

image

image

image

</details>

<details><summary>SOUL VERSION</summary>

image

image

image

image

image

</details>

**Comment by CinnamonSnowball** [comment]
> (...) so I really don't know what you're trying to argue here.

I meant the visual effect where the actual "wall" part of wall tiles is portrayed as a sloped surface on the edges of them, apologies if there's a better term for it
> Just because one sprite does it doesn't mean all of them should or that it'd look good.
We've switched to smoothed windows because non-smoothed windows look jarring and didn't fit the style of the new walls with the "sloped surface" effect

**Comment by Bm0n** [comment]
> I meant the visual effect where the actual "wall" part of wall tiles is portrayed as a sloped surface on the edges of them, apologies if there's a better term for it

Okay, I think I know what you mean now. These new walls are more TCC like than the old ones, the old shuttle walls are practically flat and have minimal shading around what can hardly be called a slope.

Just to hammer home this point here's a r_wall, new a shuttle and an old shuttle wall side by side

image


> We've switched to smoothed windows because non-smoothed windows look jarring and didn't fit the style of the new walls with the "sloped surface" effect

I tried it with smoothing, it looks bad, especially when T junction meets a window. You need that border around the window to smooth out towards the lighter transparent colour that the windows have.

**Comment by S34NW** [comment]
I think we should avoid porting flat style walls when we are generally seeing a moving trend here towards "slanted" style walls or ones that at least emulate this. (See: new rock sprites, TCC wall sprites)

These walls look like they _try_ to do this, but they are too busy on the slant for it to work well.
The windows also look super out of place since the walls are still trying to smooth with them. 

The windows would look better if you re-enabled smoothing but thickened their borders.

**Comment by S34NW** [comment]
For avoidance of doubt, I don't like the old walls either. Something needs to change, but I'm not 100% on this direction.

**Comment by Bm0n** [comment]
> I think we should avoid porting flat style walls when we are generally seeing a moving trend here towards "slanted" style walls or ones that at least emulate this. (See: new rock sprites, TCC wall sprites)

Yes

> These walls look like they _try_ to do this, but they are too busy on the slant for it to work well. That being said, they are better than the old walls in this regard.

They are slanted- at least way more slanted than the old sprites and IMO fits more in with the TCC walls. I could try making them a bit less cluttered but I'd rather not as I think personally they look really good as it.

> The windows also look super out of place since the walls are still trying to smooth with them. The windows would look better if you re-enabled smoothing but thickened their borders.

Shuttle windows were made to smooth with walls in PR #16620 , a PR which was meant to be a port of TGs smoothing system as says nothing about smoothing shuttle windows with walls. I believe this was an unintended change which I am now correcting as this is not how shuttle windows behave on TG. Block-type windows should have never smoothed with walls in the first place, it looks ugly and created monstrosities like this:
image

I am dragging my feet here but I believe it's for the best. These new windows were made with wall smoothing off in mind, you can even see it matches up with the inner grey stripe.

**Comment by S34NW** [comment]
> looks ugly and created monstrosities like this:

Yeah, this is due to the poor definition of what wall/window subtypes are used here. The nukie shuttle should have it's own type that doesn't smooth with base windows.

File: code/game/objects/structures/window.dm [hunk 1]
```diff
@@ -755,8 +755,8 @@
 	rad_insulation = RAD_NO_INSULATION
 	explosion_block = 3
 	glass_type = /obj/item/stack/sheet/plastitaniumglass
-	smoothing_groups = list(SMOOTH_GROUP_SHUTTLE_PARTS, SMOOTH_GROUP_WINDOW_FULLTILE_PLASTITANIUM, SMOOTH_GROUP_PLASTITANIUM_WALLS)
```
Comment (S34NW) [inline code/game/objects/structures/window.dm:758]: If you are going to include changes like this, please at least show them instead of lumping them in with another change with generic wording in the PR.

Comment (S34NW) [reply code/game/objects/structures/window.dm:758]: While I agree that I should have documented that more clearly (and a lot of other things that the PR changed, as holy crap it was a big one) that does not make it okay to do so now while pointing to my mistake as a reason. The smoothing refactor PR was a massive and complex PR that touched on basically every turf in the game, some things were bound to fall through the cracks.

You are supposedly touching one turf type. If you do want to change other things that is fine, but you really should document said change. Especially as it affects more than just shuttles, it touches on the entire syndicate base and how their windows smooth.

I'm not here to fight with you, I'm just saying please clearly document all the changes you make.

Comment (Bm0n) [reply code/game/objects/structures/window.dm:758]: I did say I disabled smoothing for shuttle walls on windows in my PR, I meant all shuttle windows, I didn't include a picture because it's the same effect as seen on the main shuttle.

I'll make it a little more now though.

**Comment by hal9000PR** [comment]
This pr has 3 design objections and nobody in favour after 30 days. Closing.

**Comment by Bm0n** [comment]
> This pr has 3 design objections and nobody in favour after 30 days. Closing.

the design team voted on a sprite PR?
